ThunderCloud/UIViewController+Shareable.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  UIViewController+Shareable.swift
+//  ThunderCloud
+//
+//  Created by Ben Shutt on 16/03/2021.
+//  Copyright © 2021 threesidedcube. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public extension UIViewController {
+
+    /// Source view to set `popoverPresentationController` when using iPad
+    enum ShareSourceView {
+
+        /// A `UIView`
+        case view(UIView)
+
+        /// A `UIBarButtonItem`
+        case barButtonItem(UIBarButtonItem)
+    }
+
+    /// Map `shareable` to `[ShareItems]` and share using `UIActivityViewController`
+    ///
+    /// - Parameters:
+    ///   - shareable: `Shareable` entity to share
+    ///   - defaultShareMessage: Default message to share. (Required for legacy)
+    ///   - sourceView: `ShareSourceView`
+    ///   - completionHandler: Handle completion on `UIActivityViewController`
+    func presentShare(
+        _ shareable: Shareable,
+        defaultShareMessage: String,
+        sourceView: ShareSourceView,
+        completionHandler: UIActivityViewController.CompletionWithItemsHandler? = nil
+    ) {
+        presentShare(
+            shareable.shareItems(defaultMessage: defaultShareMessage),
+            sourceView: sourceView,
+            completionHandler: completionHandler
+        )
+    }
+
+    /// Share the given `shareItems` using `UIActivityViewController`
+    ///
+    /// - Parameters:
+    ///   - shareable: `Shareable` entity to share
+    ///   - defaultShareMessage: Default message to share. (Required for legacy)
+    ///   - sourceView: `ShareSourceView`
+    ///   - completionHandler: Handle completion on `UIActivityViewController`
+    func presentShare(
+        _ shareItems: [ShareItem],
+        sourceView: ShareSourceView,
+        completionHandler: UIActivityViewController.CompletionWithItemsHandler? = nil
+    ) {
+        // Only present if there is something to share
+        guard !shareItems.isEmpty else { return }
+
+        // Create a `UIActivityViewController`
+        let shareViewController = UIActivityViewController(
+            activityItems: shareItems,
+            applicationActivities: nil
+        )
+        shareViewController.excludedActivityTypes = [
```

Comment:
Only question here is if we want to hard-code these or have them as an optional parameter with default values?

Reply:
Yeah what you suggest @simonmitchell is better 😄 ! As this PR is quite old I'm tempted to leave but that's a good suggestion